### PR TITLE
feat: create stub function for generating custom makefile

### PR DESCRIPTION
**Pull request by torresxb1**
#### Why is this change necessary?
After the prepare hook finds the SAM Metadata and links it to a specific Lambda resource, we need to create a makefile with a target/recipe for each lambda resource to be built. 

#### How does it address the issue?
Defines a stubbed function, and calls it after finding the sam metadata resources. 

#### Mandatory Checklist
**PRs will only be reviewed after checklist is complete**

- [x] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document if needed ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [x] Write/update unit tests
- [ ] Write/update integration tests
- [ ] Write/update functional tests if needed
- [x] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

File: samcli/hook_packages/terraform/hooks/prepare.py [hunk 1]
```diff
@@ -249,12 +249,35 @@ def _translate_to_cfn(tf_json: dict, output_directory_path: str, terraform_appli
...
+    Parameters
+    ----------
+    sam_metadata_resources: List[SamMetadataResource]
+        The list of sam metadata resources defined in the terraform project.
+    cfn_resources: dict
+        CloudFormation resources
+    output_directory_path: str
+        the output directory path to write the generated metadata and makefile
+    """
+    # TODO
```
Comment (torresxb1) [inline samcli/hook_packages/terraform/hooks/prepare.py:278]: will be done in the next PR

**Review by moelasmar — APPROVED** [review]

File: samcli/hook_packages/terraform/hooks/prepare.py [hunk 2]
```diff
@@ -249,12 +249,35 @@ def _translate_to_cfn(tf_json: dict, output_directory_path: str, terraform_appli
         _enrich_mapped_resources(
             sam_metadata_resources, cfn_dict["Resources"], output_directory_path, terraform_application_dir
         )
+
+        LOG.debug("Generate custom makefile for building Lambda resources")
+        _generate_custom_makefile(sam_metadata_resources, cfn_dict["Resources"], output_directory_path)
```
Comment (sriram-mv) [inline samcli/hook_packages/terraform/hooks/prepare.py:254]: nit: lets use .get() whenever we are accessing dict keys just so we dont run into key errors.

**Review by sriram-mv — APPROVED** [review]

File: tests/unit/hook_packages/terraform/test_prepare_hook.py [hunk 3]
```diff
@@ -1723,12 +1713,6 @@ def test_prepare_with_no_output_dir_path(self):
         with self.assertRaises(PrepareHookException, msg="OutputDirPath was not supplied"):
             prepare({})
 
-    def test_prepare_with_sam_metadata_resources(self):
-        pass
-
-    def test_prepare_without_sam_metadata_resources(self):
-        pass
-
```
Comment (torresxb1) [inline tests/unit/hook_packages/terraform/test_prepare_hook.py]: not used, had forgotten to remove

File: samcli/hook_packages/terraform/hooks/prepare.py (hunk 2 above)
Comment (torresxb1) [reply samcli/hook_packages/terraform/hooks/prepare.py]: done

**Pull request merged**